src/main/scala/async/listeners/locking.scala

```diff
@@ -0,0 +1,79 @@
+/** Package listeners provide some auxilliary methods to work with listeners. */
+package gears.async.listeners
+
+import gears.async._
+import Listener.{Locked, ListenerLock, Gone, PartialLock, LockMarker, LockResult}
+import scala.annotation.tailrec
+
+/** Two listeners being locked at the same time, while holding the same lock on their listener chains. */
```

Comment:
I can virtually imagine people wondering what to make out of this. Additionally, we should add a comment to channels that it is forbidden to race read and write of the same channel.

```suggestion
/** Two listeners being locked at the same time, while holding the same lock on their listener chains.
  * This happens if you attempt to lockBoth two listeners with a common downstream listener, e.g., two derived listeners of the same race.
  */
```